Wikipedia/Code/Article as a Living Document/ActivityIndicatorCollectionViewFooter.swift

```diff
@@ -0,0 +1,42 @@
+class ActivityIndicatorCollectionViewFooter: UICollectionReusableView {
+    private let loadingIndicator = UIActivityIndicatorView(style: .gray)
+    private let loadingLabel = UILabel()
+
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupView() {
+        loadingLabel.font = UIFont.wmf_font(.footnote, compatibleWithTraitCollection: traitCollection)
+        loadingLabel.text = WMFLocalizedString("loading-indicator-text", value: "Loading", comment: "Text shown underneath loading indicator.").uppercased()
+
+        let stackView = UIStackView(arrangedSubviews: [loadingIndicator, loadingLabel])
+        stackView.axis = .vertical
+        stackView.spacing = 4.0
+        stackView.alignment = .center
+
+        addSubview(stackView)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
```

Comment:
I see some layout constraint errors on iOS13 simulators (14 seems fine). The UI lays out fine though, and they only seem to show when cells are appended to the collection view, not when the footer itself is displayed. So maybe it's just a matter of changing the priority on one of these to 999 for some state that it gets in when new cells are appended. Everything else looks great!

Reply:
Good catch. This is now fixed!